modin/core/dataframe/pandas/dataframe/dataframe.py

```diff
@@ -990,6 +990,22 @@ def _take_2d_positional(
         if col_positions is None and row_positions is None:
             return self.copy()
 
+        if col_positions is None and row_positions is not None:
+            # This way tries to eliminate the optimization that first takes part of the data using
+            # the mask operation, so that later less data is concated into a whole column.
+            # Note: in the case where a small part of the data is discarded, the overhead of the
+            # engines (for putting data in and out of storage) can exceed the resulting speedup.
+            all_rows = None
+            if self.has_materialized_index:
+                all_rows = len(self.index)
+            elif self._row_lengths_cache:
+                all_rows = sum(self._row_lengths_cache)
+            if all_rows:
+                if len(row_positions) > int(0.9 * all_rows):
```

Comment:
```suggestion
                if len(row_positions) > 0.9 * all_rows:
```
in Python, there's no need to cast to `int` to compare 🙃 

Also I wonder if the threshold should be tuned - did you try different coefficients?